components/post_view/post_list.jsx

```diff
@@ -173,24 +178,61 @@ export default class PostList extends React.PureComponent {
-        let newPostListIds;
-
-        if (state.atEnd) {
-            newPostListIds = [...postListIds, PostListRowListIds.CHANNEL_INTRO_MESSAGE];
-        } else if (props.postVisibility >= Constants.MAX_POST_VISIBILITY) {
-            newPostListIds = [...postListIds, PostListRowListIds.MAX_MESSAGES_LOADED];
-        } else if (state.autoRetryEnable) {
-            newPostListIds = [...postListIds, PostListRowListIds.MORE_MESSAGES_LOADER];
-        } else {
-            newPostListIds = [...postListIds, PostListRowListIds.MANUAL_TRIGGER_LOAD_MESSAGES];
+        const {prevPropsPostListIds, postListIds: statePostListIds} = state;
+        const propsPostListIds = props.postListIds;
+        let postListIds = props.postListIds || [];
+
+        if (propsPostListIds) {
+            const hasPostListIdsChanged = prevPropsPostListIds && (prevPropsPostListIds !== propsPostListIds);
+
+            if (!prevPropsPostListIds) {
+                const sliceIndexForPosts = props.focusedPostId ? MAXIMUM_POSTS_FOR_SLICING.permalink : MAXIMUM_POSTS_FOR_SLICING.channel;
+                postListIds = postListIds.slice(0, sliceIndexForPosts);
+            } else if (hasPostListIdsChanged) {
+                if (statePostListIds[0] !== propsPostListIds[0]) {
+                    // Post added at bottom
+                    const oldestPostId = PostList.getOldestVisiblePostId(statePostListIds);
+                    const oldestIdIndexInProps = propsPostListIds.findIndex((postId) => postId === oldestPostId);
+                    postListIds = propsPostListIds.slice(0, oldestIdIndexInProps);
+                } else if (statePostListIds[0] === propsPostListIds[0]) {
+                    //Bottom most post is still the same but there is a change in length of postsIds array
+                    const oldestPostIdFromState = PostList.getOldestVisiblePostId(propsPostListIds);
+                    const oldestPostIdFromPresentProps = PostList.getOldestVisiblePostId(prevPropsPostListIds);
+                    if (oldestPostIdFromState === oldestPostIdFromPresentProps) {
+                        // top most message and bottom most message hasnt changed but length changed.
+                        // Happens for cases like deleted message
+                        const changeInLength = propsPostListIds.length - prevPropsPostListIds.length;
+                        postListIds = propsPostListIds.slice(0, statePostListIds.length + changeInLength);
+                    }
+                }
+            } else {
+                // no change in propsPostListIds but could have a change in state that needs to be updated
+                const topPostId = PostList.getTopIdForPostList(state);
+                if (statePostListIds[state.postListIds.length - 1] === topPostId) {
+                    return null;
+                }
+            }
         }
 
+        const topPostIdToBeAdded = PostList.getTopIdForPostList(state);
```

Comment:
Does the virtualized list have anything like the `ListHeaderComponent` and `ListFooterComponent` props that RN's `FlatList` has? It would be really nice if we didn't have to add this onto the PostList every time